react/Popover/Popover.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react'
+import ReactDOM from 'react-dom'
+import PropTypes from 'prop-types'
+export default class Popover extends Component {
+  constructor(props) {
+    super(props)
+    this.element = ''
+    this.state = {
+      style: { display: 'block', visibility: 'hidden', position: 'fixed' }
+    }
+  }
+
+  //When the component is ready, than let's do some Maths
+  componentDidMount() {
+    requestAnimationFrame(this.computeCoordinates.bind(this))
+  }
+  computeCoordinates() {
+    const { style } = this.props
+    //Find me & give me my coordinates
+    // eslint-disable-next-line
+    const domItem = ReactDOM.findDOMNode(this)
+    const domItemCoordinates = domItem.getBoundingClientRect()
+    // eslint-disable-next-line
+    const menuElement = domItem.parentNode
+    const scrollTop = this.getScrollParent(menuElement)
+    const coordinates = menuElement.getBoundingClientRect()
+
+    //If we have don't have enough space => maths
+    const hasNotEnoughSpace =
+      domItemCoordinates.bottom - scrollTop > window.innerHeight
+    this.setState({
+      style: {
+        ...style,
+        top: hasNotEnoughSpace
+          ? coordinates.top - domItemCoordinates.height
+          : coordinates.top + coordinates.height,
+        left: coordinates.left + coordinates.width - domItemCoordinates.width,
+        position: 'fixed',
+        display: 'block',
+        visibility: 'visible'
+      }
```

Comment:
I think it would be clearer to just set the necessary style attributes in the state and not copy the whole thing.

```
this.setState({ offsetTop: ..., offsetLeft: ..., visibility: ... })
```